doc/neps/nep-0029-deprecation_policy.rst

```diff
@@ -0,0 +1,186 @@
+==========================================================================================
+NEP 29 — A standard community policy for dropping support of old Python and NumPy versions
+==========================================================================================
+
+
+:Author: Thomas A Caswell <tcaswell@gmail.com>, Andreas Mueller, Brian Granger, Madicken Munk, Ralf Gommers, Matt Haberland <mhaberla@calpoly.edu>, Matthias Bussonnier <bussonniermatthias@gmail.com>, Stefan van der Walt
+:Status: Draft
+:Type: Informational Track
+:Created: 2019-07-13
+
+
+Abstract
+--------
+
+All projects across the ecosystem should adopt a common time window
+based policy for increasing the minimum version of Python and NumPy
+that downstream projects support.  By standardizing this policy
+across the community we will make it easier for downstream projects to
+plan.
+
+This is an unusual NEP in that it is recommendations for community-wide
+policy and not for changes to NumPy itself.  However, we do not
+currently have a place for SPEEPs (Scientific Python Ecosystem
+Enhancement Proposals) and given NumPy's central role in our community
+this seems like the proper place to document this.
+
+
+This is being put forward by maintainers of Matplotlib, scikit-learn,
+IPython, Jupyter, yt, SciPy, NumPy, and scikit-image.
+
+
+
+Detailed description
+--------------------
+
+When making a new major or minor release, projects should support all
+minor versions of ``Python`` initially released in the prior 42
```

Comment:
How does this support algorithm handle the release of 4.0.0

Reply:
I think if we get kernel-style major version bumps, we can treat it as any other minor release and I don't think there is upstream appetite for a 2->3 style 3->4 transition.  Not sure how to phrase that succinctly though.

Reply:
I suggest to just put a disclaimer in about major versions, since it's not really possible to predict in advance what the right strategy is. It really depends on what breaks in 4.0 and how the community reacts to that.

Reply:
Added a sentence at the end of this section.